axelor-crm/src/main/resources/views/Opportunity.xml

```diff
@@ -231,7 +232,11 @@
     	<action name="action-opportunity-set-closed-lost"/>
     	<action name="save"/>
     </action-group>
-    
+
+  <action-attrs name="action-crm-opportunity-attrs-lost-reason-onchange">
+    <attribute name="hidden" for="lostReasonStr" expr="eval: !lostReason?.freeText"/>
```

Comment:
Use hideIf with dotted hidden field, instead of creating action and calling it on onChange.